### PR TITLE
merge systemd files for stash cache & origin (SOFTWARE-5028)

### PR DESCRIPTION
**Pull request by edquist**
I am a little fuzzy on where all i need `stash-authfile@cache` vs `stash-authfile@%i` vs `stash-authfile@`.  Hopefully I got it right but please double check  :)

### CODE REVIEW

**Comment by brianhlin** [comment]
As an aside, the systemd man pages are really solid but spread across a bunch of different pages. You can find info on the specifiers in `man systemd.unit`

File: configs/stash-cache/config.d/50-stash-cache-authz.cfg [hunk 1]
```diff
@@ -33,15 +33,15 @@ if defined ?StashCacheAuthfile && named stash-cache-auth
    acc.authdb $StashCacheAuthfile
 # nor negations ("!")
 else if named stash-cache-auth
-   # This authfile is automatically generated by stash-cache-authfile.service
+   # This authfile is automatically generated by stash-authfile@cache.service
```
Comment (matyasselmeci) [inline configs/stash-cache/config.d/50-stash-cache-authz.cfg]:
```suggestion
   # This authfile is automatically generated by stash-authfile@stash-cache-auth.service
```

File: configs/stash-cache/config.d/50-stash-cache-authz.cfg [hunk 2]
```diff
@@ -33,15 +33,15 @@ if defined ?StashCacheAuthfile && named stash-cache-auth
...
    acc.authdb /run/stash-cache-auth/Authfile
 fi
 
 if defined ?StashCachePublicAuthfile && named stash-cache
    acc.authdb $StashCachePublicAuthfile
 else if named stash-cache
    # The unauthenticated instance uses a separate auth file from the
-   # authenticated instance; file is generated by stash-cache-authfile.service
+   # authenticated instance; file is generated by stash-authfile@cache.service
```
Comment (matyasselmeci) [inline configs/stash-cache/config.d/50-stash-cache-authz.cfg]:
```suggestion
   # authenticated instance; file is generated by stash-authfile@stash-cache.service
```

File: configs/stash-cache/overrides/10-stash-cache-overrides.conf [hunk 3]
```diff
@@ -8,6 +8,6 @@
 # - Depends on fetch-crl and the reporter script, but doesn't need these
 #   to finish or succeed for startup.
 
-Wants=stash-cache-authfile.service stash-cache-authfile.timer \
+Wants=stash-authfile@%i.service stash-authfile@%i.timer \
       xrootd-renew-proxy.timer xcache-reporter.timer
-After=stash-cache-authfile.service xrootd-renew-proxy.service
+After=stash-authfile@%authfile@%i.service xrootd-renew-proxy.service
```
Comment (matyasselmeci) [inline configs/stash-cache/overrides/10-stash-cache-overrides.conf]:
```suggestion
After=stash-authfile@%i.service xrootd-renew-proxy.service
```

**Review by matyasselmeci — APPROVED** [review]
LGTM except for a few fixups; pre-approving.

File: configs/stash-cache/overrides/10-stash-cache-overrides.conf (hunk 3 above)
Comment (edquist) [reply configs/stash-cache/overrides/10-stash-cache-overrides.conf]: Derp.

**Pull request merged**